skbio/alignment/_tabular_msa.py

```diff
@@ -741,6 +741,61 @@ def consensus(self):
                      positional_metadata=positional_metadata)
 
     @experimental(as_of='0.4.0-dev')
+    def conservation(self, metric, nan_on_degenerate=False, nan_on_gap=True):
+        """Apply metric to compute the position conservation for the alignment
+
+        Parameters
+        ----------
+        metric : callable
+            Metric that should be applied for computing conservation. If
+            callable, must take a string representing the characters at single
+            position in the alignment, and the TabularMSA object and return a
+            single value.
+        nan_on_degenerate : bool, optional
+            If ``True``, positions in the resulting array that correspond to
+            alignment positions containing degenerate characters will have
+            np.nan as their value.
+        nan_on_gap : bool, optional
+            If ``True``, positions in the resulting array that correspond to
+            alignment positions with gaps will have np.nan as their value.
+
+        Returns
+        -------
+        np.array
+            Value of applying ``metric`` to each position in the alignment.
+
+        Raises
+        ------
+        KeyError
```

Comment:
I think a `ValueError` would be more appropriate as this is thrown when a value contained in the position is not supported under this condition.